renku/service/jobs/cleanup.py

```diff
@@ -21,61 +21,107 @@
-            created_at = (file.created_at -
-                          datetime.utcfromtimestamp(0)).total_seconds() * 1e+3
+        jobs = [
+            job for job in cache.get_jobs(user) if job.state in
+            [USER_JOB_STATE_ENQUEUED, USER_JOB_STATE_IN_PROGRESS]
+        ]
 
-            if not file.created_at:
-                created_at = (time.time() - ttl) * 1e+3
+        for file in files:
+            if file_cleanup_lock_check(jobs, file):
+                continue
 
-            old = ((time.time() * 1e+3) - created_at) / 1e+3
+            ttl_check = file_cleanup_ttl_check(file)
+            file_path = file.abs_path
 
-            if old >= ttl:
-                file_path = file.abs_path
-                if file_path.exists() and file_path.is_file():
+            if ttl_check:
+                if file_path.is_file():
                     try:
                         file_path.unlink()
                     except FileNotFoundError:
                         pass
 
-                if file_path.exists() and file_path.is_dir():
+                if file_path.is_dir():
                     shutil.rmtree(str(file_path))
 
                 file.delete()
 
 
+def project_cleanup_ttl_check(project, ttl=None):
+    """Check if file should be deleted."""
+    ttl = ttl or int(os.getenv('RENKU_SVC_CLEANUP_TTL_PROJECTS', 1800))
+    # If record does not contain created_at,
+    # it means its an old record, and we should mark it for deletion.
+    created_at = (project.created_at -
+                  datetime.utcfromtimestamp(0)).total_seconds() * 1e+3
+
+    if not project.created_at:
+        created_at = (time.time() - ttl) * 1e+3
+
+    old = ((time.time() * 1e+3) - created_at) / 1e+3
+
+    return old >= ttl and project.abs_path.exists()
+
+
+def project_cleanup_lock_check(jobs, project):
+    """Check if project should be deleted."""
+    for job in jobs:
+        if project.project_id in job.locked:
+            return True
+
+    return False
+
+
 def cache_project_cleanup():
     """Cache project a cleanup job."""
-    ttl = int(os.getenv('RENKU_SVC_CLEANUP_TTL_PROJECTS', 1800))
-    cache = ProjectManagementCache()
+    cache = ServiceCache()
 
     for user, projects in cache.user_projects():
-        for project in projects:
-
-            # If record does not contain created_at,
-            # it means its an old record, and we should mark it for deletion.
-            created_at = (project.created_at -
-                          datetime.utcfromtimestamp(0)).total_seconds() * 1e+3
+        jobs = [
+            job for job in cache.get_jobs(user) if job.state in
+            [USER_JOB_STATE_ENQUEUED, USER_JOB_STATE_IN_PROGRESS]
+        ]
 
-            if not project.created_at:
-                created_at = (time.time() - ttl) * 1e+3
-
-            old = ((time.time() * 1e+3) - created_at) / 1e+3
+        for project in projects:
+            if project_cleanup_lock_check(jobs, project):
```

Comment:
Small nitpick but could we flip the logic in this method around?

We have
```
def project_cleanup_ttl_check(project, ttl=None):
    """Check if file should be deleted."""
```

where `True` means "delete the project"

and
```
def project_cleanup_lock_check(jobs, project):
    """Check if project should be deleted."""
```
where `True` means "Don't delete the project", which is a bit inconsistent.